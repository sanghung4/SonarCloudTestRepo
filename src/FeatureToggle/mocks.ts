import { Feature } from 'generated/graphql';

export const mockFeatures: Feature[] = [
  { id: '1', isEnabled: false, name: 'noooo' },
  { id: '2', isEnabled: true, name: 'fine' },
  { id: '3', isEnabled: true, name: 'test' }
];
