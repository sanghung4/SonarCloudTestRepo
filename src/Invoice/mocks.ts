import { GetInvoiceDocument } from 'generated/graphql';

export const successMincron = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '123456',
      invoiceNumber: '5460026'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5460026',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: '40251/13325',
        invoiceDate: '10/28/2020',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'AVLNEP7',
        jobName: 'NEP 7 ASHEVILLE',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2020',
        deliveryMethod: 'Our Truck',
        subtotal: '3184.28',
        tax: '262.76',
        otherCharges: '0.0',
        paidToDate: '1000.00',
        invoiceItems: [
          {
            id: '1738433F4-91A7-4637-8C60-B4DF59A060C1',
            brand: 'Advanced Drainage Systems, Inc.',
            name: '120 x 7.5, 100 Sq Yd, Geosynthetic, Double, Photodegradable, Erosion Control Blanket',
            partNumber: '8X1125STRAWDN',
            mfr: '00S2TT',
            thumb: 'null',
            price: '43.59',
            qty: {
              quantityOrdered: 1,
              quantityShipped: 1
            }
          }
        ]
      }
    }
  }
};

export const successMincron_5448540 = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '91492',
      invoiceNumber: '5448540'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5282411',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: 'WILKESBORO JIM',
        invoiceDate: '10/06/2021',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'SHOP',
        jobName: 'SHOP',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2021',
        deliveryMethod: 'Our Truck',
        subtotal: '2184.28',
        tax: '162.73',
        otherCharges: '0.0',
        paidToDate: '139.91',
        invoiceItems: {
          id: 'F02FD85F-4DCE-4EA7-8459-2DA7CFD689DC',
          brand: 'Certex USA, name:2 x 12, Nylon, Eye and Eye, Ply Sling',
          partNumber: 'SLING212,mfr:CX07-0126-12',
          thumb:
            'https://images.tradeservice.com/ProductImages/DIR100193/CERTEX_CX07-0126-12_LINE_SML.jpg',
          price: '40.00',
          qty: {
            quantityOrdered: 1,
            quantityShipped: 1
          }
        }
      }
    }
  }
};
