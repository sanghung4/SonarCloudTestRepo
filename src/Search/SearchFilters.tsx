import { Dispatch, useCallback, useMemo } from 'react';

import {
  Box,
  Button,
  CircularProgress,
  Drawer,
  IconButton,
  Typography,
  alpha,
  useScreenSize
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';

import useSearchQueryParams from 'Search/util/useSearchQueryParams';
import { getSelectedFiltersArray } from 'Search/util/searchFiltersUtil';
import { AggregationResults, Maybe } from 'generated/graphql';
import { CloseIcon } from 'icons';
import SearchFiltersContent from './SearchFiltersContent';

/**
 * Types
 */
export type Props = {
  loading: boolean;
  count: number;
  filters?: Maybe<AggregationResults>;
  filtersOpen: boolean;
  onShowHide: Dispatch<boolean>;
};

export default function SearchFilters(props: Props) {
  /**
   * Custom Hooks
   */
  const { isSmallScreen } = useScreenSize();
  const { t } = useTranslation();
  const [params, setParams] = useSearchQueryParams();
  const { filters = [], criteria: searchTerm } = params;

  /**
   * Memos
   */
  const selectedFilters = useMemo(selectedFiltersMemo, [filters]);

  /**
   * Callbacks
   */
  const handleClear = useCallback(handleClearCb, [searchTerm, setParams]);

  return isSmallScreen ? (
    <Drawer
      PaperProps={{ sx: { bottom: 0 } }}
      anchor="top"
      open={props.filtersOpen}
      onClose={() => props.onShowHide(false)}
    >
      <Box display="flex" flexDirection="column" minHeight={1}>
        <Box
          px={3}
          pt={5.5}
          pb={1.5}
          display="flex"
          alignItems="center"
          sx={{
            borderBottom: 1,
            borderBottomColor: 'lighterGray.main'
          }}
        >
          <Box flex="1" />
          <Typography
            variant="h5"
            component="span"
            color="primary"
            sx={{ fontWeight: 700 }}
          >
            {t('common.filter')}
          </Typography>
          <Box display="flex" flex="1" justifyContent="flex-end">
            <IconButton
              onClick={() => props.onShowHide(false)}
              size="large"
              sx={{ p: 0 }}
            >
              <CloseIcon data-testId="close-icon" />
            </IconButton>
          </Box>
        </Box>
        <Box flex="1" overflow="auto">
          <SearchFiltersContent
            loading={props.loading}
            filters={props.filters}
            onClear={handleClear}
          />
        </Box>
        <Box
          display="flex"
          p={4}
          sx={(theme) => ({
            borderTop: 1,
            borderTopColor: 'lightGray.main',
            boxShadow: `0 -2px 4px 0 ${alpha(theme.palette.common.black, 0.15)}`
          })}
        >
          <Button
            data-testId="search-filter-button"
            onClick={() => props.onShowHide(false)}
            sx={{ flex: 1 }}
          >
            {`${t('search.showResults')} (`}
            {props.loading ? (
              <CircularProgress
                size="1em"
                sx={{
                  display: 'flex',
                  my: 0,
                  mx: 0.5,
                  '& .MuiCircularProgressReece-bottom': {
                    color: 'primary.main'
                  }
                }}
              />
            ) : (
              props.count
            )}
            {')'}
          </Button>
          {selectedFilters.length > 0 ? (
            <Button variant="text" onClick={handleClear} sx={{ ml: 3 }}>
              {t('common.clear')}
            </Button>
          ) : null}
        </Box>
      </Box>
    </Drawer>
  ) : (
    <SearchFiltersContent
      loading={props.loading}
      filters={props.filters}
      onClear={handleClear}
    />
  );

  /**
   * Memo defs
   */
  function selectedFiltersMemo() {
    return getSelectedFiltersArray(filters);
  }

  /**
   * Callback defs
   */
  function handleClearCb() {
    setParams({
      criteria: searchTerm
    });
  }
}
