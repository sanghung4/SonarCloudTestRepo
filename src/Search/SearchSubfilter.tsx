import {
  Checkbox,
  ListItem,
  ListItemIcon,
  ListItemText,
  Skeleton,
  Typography
} from '@dialexa/reece-component-library';
import { kebabCase } from 'lodash-es';

import {
  AggregationItem,
  AggregationResults,
  ProductAttribute
} from 'generated/graphql';

/**
 * Types
 */
type SearchSubfilterProps = {
  checked: boolean;
  filter: keyof AggregationResults;
  handleToggle: (changedFilter: ProductAttribute) => void;
  subfilter: AggregationItem;
  stockText: string;
  loading: boolean;
  filterDisabled: boolean;
};

/**
 * SearchSubFilter Component
 */
export default function SearchSubfilter(props: SearchSubfilterProps) {
  /**
   * Callback
   */
  const listItemClickCb = () => {
    if (!props.filterDisabled) {
      props.handleToggle({
        attributeType: props.filter,
        attributeValue: props.subfilter.value || ''
      });
    }
  };

  const styleCheckbox = {
    cursor: props.filterDisabled ? 'not-allowed' : 'pointer',
    p: 0,
    ml: props.filterDisabled ? '2px' : 0,
    mr: props.filterDisabled ? '4px' : 0,
    backgroundColor: props.filterDisabled ? '#EEEEEE' : '',
    height: props.filterDisabled ? '1.1em' : '',
    width: props.filterDisabled ? '1.2em' : 'auto',
    borderRadius: props.filterDisabled ? '5%' : '50%',
    left: props.filterDisabled ? '0.1em' : ''
  };

  /**
   * Render
   */
  return (
    <ListItem
      key={props.subfilter.value}
      role={undefined}
      onClick={listItemClickCb}
      sx={{
        cursor: props.filterDisabled ? 'not-allowed' : 'pointer',
        px: 0,
        py: 1,
        '& .MuiListItemIcon-root, & .MuiListItemText-root': {
          mb: 0
        },
        '&:hover': {
          bgcolor: 'transparent'
        }
      }}
      data-testid={kebabCase(`filter-line-item-${props.subfilter.value}`)}
    >
      <ListItemIcon
        sx={{
          minWidth: 'auto',
          mr: 1,
          display: 'flex',
          flexDirection: 'column'
        }}
      >
        <Checkbox
          checked={props.checked}
          color="primary"
          tabIndex={-1}
          sx={styleCheckbox}
          data-testid={kebabCase(`checkbox-${props.subfilter.value}`)}
        />
      </ListItemIcon>
      <ListItemText sx={{ my: 0 }}>
        {props.filter === 'inStockLocation' && props.loading ? (
          <Skeleton variant="rectangular" width="100%" />
        ) : (
          <>
            <Typography
              variant="caption"
              color={props.filterDisabled ? 'secondary02.main' : 'textPrimary'}
            >
              {props.stockText || props.subfilter.value}
              <Typography
                variant="caption"
                component="span"
                color="textPrimary"
                fontWeight={300}
                ml={0.5}
              >
                ({props.subfilter.count})
              </Typography>
            </Typography>
          </>
        )}
      </ListItemText>
    </ListItem>
  );
}
