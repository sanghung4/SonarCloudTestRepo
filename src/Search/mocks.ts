export const PRODUCT = {
  id: 'MSC-43443',
  partNumber: '43443',
  name: 'Br 1/2 Pex X 1/2 Cop Adpt',
  manufacturerName: 'Uponor',
  manufacturerNumber: 'LF4515050',
  price: 3.541,
  stock: {
    homeBranch: {
      branchName: 'TEXARKANA',
      branchId: '1023',
      address1: null,
      address2: null,
      city: null,
      state: null,
      zip: null,
      phone: null,
      distanceToBranch: null,
      availability: 84,
      __typename: 'StoreStock'
    },
    otherBranches: [
      {
        branchName: 'ABILENE PLBG',
        branchId: '1001',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 622,
        __typename: 'StoreStock'
      },
      {
        branchName: 'AMARILLO',
        branchId: '1002',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 0,
        __typename: 'StoreStock'
      },
      {
        branchName: 'WICHITA FLLS',
        branchId: '1006',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 19,
        __typename: 'StoreStock'
      },
      {
        branchName: 'LUBBOCK',
        branchId: '1008',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 51,
        __typename: 'StoreStock'
      },
      {
        branchName: 'ODESSA',
        branchId: '1009',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 20,
        __typename: 'StoreStock'
      },
      {
        branchName: 'BIG SPRING',
        branchId: '1010',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 9,
        __typename: 'StoreStock'
      },
      {
        branchName: 'MIDLAND SHOW',
        branchId: '1011',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 0,
        __typename: 'StoreStock'
      },
      {
        branchName: '*KILGORE',
        branchId: '1021',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 0,
        __typename: 'StoreStock'
      },
      {
        branchName: 'TYLER',
        branchId: '1022',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 224,
        __typename: 'StoreStock'
      },
      {
        branchName: '*NACOGDOCHES',
        branchId: '1024',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 0,
        __typename: 'StoreStock'
      },
      {
        branchName: 'LONGVW PLBG',
        branchId: '1027',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 111,
        __typename: 'StoreStock'
      },
      {
        branchName: 'SAN ANGELO',
        branchId: '1029',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 82,
        __typename: 'StoreStock'
      },
      {
        branchName: 'MIDLAND PLBG',
        branchId: '1032',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 0,
        __typename: 'StoreStock'
      },
      {
        branchName: 'CARLSBAD',
        branchId: '1035',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 0,
        __typename: 'StoreStock'
      },
      {
        branchName: 'ALAMOGORDO',
        branchId: '1039',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 0,
        __typename: 'StoreStock'
      },
      {
        branchName: 'LAS CRUCES',
        branchId: '1040',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 26,
        __typename: 'StoreStock'
      },
      {
        branchName: 'ALBUQUERQUE',
        branchId: '1044',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 268,
        __typename: 'StoreStock'
      },
      {
        branchName: 'MCALLEN',
        branchId: '1045',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 0,
        __typename: 'StoreStock'
      },
      {
        branchName: 'BROWNSVILLE',
        branchId: '1055',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 0,
        __typename: 'StoreStock'
      },
      {
        branchName: 'HARLINGEN',
        branchId: '1056',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 49,
        __typename: 'StoreStock'
      },
      {
        branchName: 'GARDEN CITY',
        branchId: '1075',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 138,
        __typename: 'StoreStock'
      },
      {
        branchName: 'HUTCHINSON',
        branchId: '1076',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 37,
        __typename: 'StoreStock'
      },
      {
        branchName: 'LUBBOCK SHOW',
        branchId: '1079',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 0,
        __typename: 'StoreStock'
      },
      {
        branchName: 'LAREDO',
        branchId: '1084',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 0,
        __typename: 'StoreStock'
      },
      {
        branchName: 'ABILENE SHOW',
        branchId: '1085',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 0,
        __typename: 'StoreStock'
      },
      {
        branchName: 'EL PASO',
        branchId: '1100',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 26,
        __typename: 'StoreStock'
      },
      {
        branchName: 'SANTA FE',
        branchId: '1109',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 364,
        __typename: 'StoreStock'
      },
      {
        branchName: 'EL PASO SHOW',
        branchId: '1112',
        address1: null,
        address2: null,
        city: null,
        state: null,
        zip: null,
        phone: null,
        distanceToBranch: null,
        availability: 0,
        __typename: 'StoreStock'
      }
    ],
    __typename: 'Stock'
  },
  taxonomy: null,
  categories: [
    'Pipe & Fittings',
    'Plastic Pipe & Fittings - Supply',
    'PEX Pipe & Fittings'
  ],
  technicalDocuments: [
    {
      name: 'MFR Catalog Page',
      url: 'http://images.tradeservice.com/ATTACHMENTS/DIR100147/UPONORE00439_101.pdf',
      __typename: 'TechDoc'
    },
    {
      name: 'Technical Specification',
      url: 'http://images.tradeservice.com/ATTACHMENTS/DIR100147/UPONORE00450_1.pdf',
      __typename: 'TechDoc'
    },
    {
      name: 'Installation Sheet',
      url: 'http://images.tradeservice.com/INSTALLATIONSHEETS/DIR100005/UPONORINST00041.pdf',
      __typename: 'TechDoc'
    }
  ],
  environmentalOptions: ['Low lead compliant'],
  upc: '67337223245',
  unspsc: '40183109',
  seriesModelFigureNumber: null,
  productOverview:
    'Adapter Fitting; Type Transition, Straight; Size A 1/2"; Size B 1/2"; End Connection PEX x Female Soldered; Material C69300 Brass; Material Specification ASTM F877/F1960; Application Hot and Cold Domestic Potable Water System, Hydronic Radiant Heating and Cooling System, Residential Fire Sprinkler System; Applicable Standard NSF 14/61/372, CAN/CSA B137.5, UPC, IAPMO, IBC, IMC, IRC, IPC, NSPC, NPC, UMC, ICC ES PMG 1006/1012, HUD MR 1269, UL 1821, ULC/ORD C199P; Packaging Quantity 25 per Pack;...',
  featuresAndBenefits:
    'Lead Free; Brass Sweat Adapters Transition Uponor PEX Tubing to Copper Pipe for Use in Hot and Cold Domestic Potable Water Systems, Residential Fire Safety and Hydronic Radiant Heating and Cooling Systems; One these Adapters are Safe for Direct Burial in Soil; 25 Year Limited Warranty',
  techSpecifications: [
    {
      name: 'material',
      value: 'C69300 Brass',
      __typename: 'TechSpec'
    },
    {
      name: 'Applicable Standard',
      value:
        'NSF 14/61/372, CAN/CSA B137.5, UPC, IAPMO, IBC, IMC, IRC, IPC, NSPC, NPC, UMC, ICC ES PMG 1006/1012, HUD MR 1269, UL 1821, ULC/ORD C199P',
      __typename: 'TechSpec'
    },
    {
      name: 'Application',
      value:
        'Hot and Cold Domestic Potable Water System, Hydronic Radiant Heating and Cooling System, Residential Fire Sprinkler System',
      __typename: 'TechSpec'
    },
    {
      name: 'End Connection',
      value: 'PEX x Female Soldered',
      __typename: 'TechSpec'
    },
    {
      name: 'Material',
      value: 'C69300 Brass',
      __typename: 'TechSpec'
    },
    {
      name: 'Material Specification',
      value: 'ASTM F877/F1960',
      __typename: 'TechSpec'
    },
    {
      name: 'Size A',
      value: '1/2"',
      __typename: 'TechSpec'
    },
    {
      name: 'Size B',
      value: '1/2"',
      __typename: 'TechSpec'
    },
    {
      name: 'Type',
      value: 'Transition, Straight',
      __typename: 'TechSpec'
    }
  ],
  imageUrls: {
    thumb:
      'http://images.tradeservice.com/ProductImages/DIR100108/UPONOR_LF4512525_SML.jpg',
    small:
      'http://images.tradeservice.com/ProductImages/DIR100108/UPONOR_LF4512525_SML.jpg',
    medium:
      'http://images.tradeservice.com/ProductImages/DIR100108/UPONOR_LF4512525_MED.jpg',
    large:
      'http://images.tradeservice.com/ProductImages/DIR100107/UPONOR_LF4512525_LRG.jpg',
    __typename: 'ImageUrls'
  },
  packageDimensions: null,
  minIncrementQty: 0,
  erp: 'ECLIPSE',
  __typename: 'Product'
};
