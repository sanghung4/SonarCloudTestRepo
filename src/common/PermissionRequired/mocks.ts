import { UserContextType } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';

export const authConfigNoPerm = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
export const authConfig = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    userId: 'test',
    permissions: [Permission.EDIT_LIST, Permission.APPROVE_CART],
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
