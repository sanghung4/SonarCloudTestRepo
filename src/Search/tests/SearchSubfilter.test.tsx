import SearchSubfilter from 'Search/SearchSubfilter';
import { FILTERS, FILTER_ARRAY } from 'Search/SearchFilters.mocks';
import { render } from 'test-utils/TestWrapper';

type filterType = keyof typeof FILTERS;
let filterData: filterType = 'inStockLocation';

const SearchSubfilterProps = {
  checked: false,
  filter: filterData,
  handleToggle: () => {},
  subfilter: FILTER_ARRAY[0],
  stockText: 'In-stock @ Morrison Supply',
  loading: false,
  filterDisabled: true
};

describe('Check isPricingOnly branches', () => {
  it('Check branch matches snapshot post load', async () => {
    const { container } = render(<SearchSubfilter {...SearchSubfilterProps} />);

    expect(container).toMatchSnapshot();
  });

  it('Check in-stock filter to be enabled when branch is not isPricingOnly', async () => {
    SearchSubfilterProps.filterDisabled = false;
    const { findByTestId } = render(
      <SearchSubfilter {...SearchSubfilterProps} />
    );

    const stkFilterElement = await findByTestId(
      'filter-line-item-mock-filter-1'
    );

    expect(stkFilterElement).not.toHaveClass('Mui-disabled');
  });
});
