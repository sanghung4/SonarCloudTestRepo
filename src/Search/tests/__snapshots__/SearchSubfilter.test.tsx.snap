// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check isPricingOnly branches Check branch matches snapshot post load 1`] = `
<div>
  <li
    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-s7nh9p-MuiListItem-root"
    data-testid="filter-line-item-mock-filter-1"
  >
    <div
      class="MuiListItemIcon-root css-18jpnm2-MuiListItemIcon-root"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12gbesn-MuiButtonBase-root-MuiCheckbox-root"
        data-testid="checkbox-mock-filter-1"
      >
        <input
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          tabindex="-1"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
    </div>
    <div
      class="MuiListItemText-root css-yuf5hv-MuiListItemText-root"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-zvlsvz-MuiTypography-root"
      >
        <span
          class="MuiTypography-root MuiTypography-caption css-rk7jif-MuiTypography-root"
        >
          In-stock @ Morrison Supply
          <span
            class="MuiTypography-root MuiTypography-caption css-eospm3-MuiTypography-root"
          >
            (
            100
            )
          </span>
        </span>
      </span>
    </div>
  </li>
</div>
`;
